src/pip/_internal/resolution/resolvelib/factory.py

```diff
@@ -288,54 +346,44 @@ def find_candidates(
         # can be made quicker by comparing only the id() values.
         incompat_ids = {id(c) for c in incompatibilities.get(identifier, ())}
 
+        # Collect basic lookup information from the requirements.
         explicit_candidates = set()  # type: Set[Candidate]
         ireqs = []  # type: List[InstallRequirement]
         for req in requirements[identifier]:
             cand, ireq = req.get_candidate_lookup()
-            if cand is not None and id(cand) not in incompat_ids:
+            if cand is not None and id(cand):
```

Comment:
@uranusjr why `and id(cand)` ?

Reply:
Oops, I wanted to delete the entire `and ...` clause (it is now handled elsewhere). Will fix this.